r/R/dotplot.R

```diff
@@ -0,0 +1,34 @@
+runDotPlot <- function(req, data) {
+  type <- req$body$type
+  cellSets <- req$body$cellSets$children
+  data$custom <- NA
+
+  object_ids <- data$cells_id
+  for (i in seq_along(cellSets)) {
+    set <- cellSets[[i]]
+    filtered_cells <- intersect(set$cellIds, object_ids)
+    data$custom[object_ids %in% filtered_cells] <- i
+  }
```

Comment:
This looks really good to me. I have two suggestions:

1. We agree to use `camelCase` for function names and `snake_case` or `dot.case`(?) for variables.
2. Minor change: I rearranged the destructuring and added a comment mainly just to keep straight that `type` is not being destructured for the action we are about to perform.

```suggestion
  type <- req$body$type
  
  # construct clusters from cell sets
  data$custom <- NA
  cells_id <- data$cells_id
  cell_sets <- req$body$cellSets$children
  
  for (i in seq_along(cell_sets)) {
      set <- cell_sets[[i]]
      filtered_cells <- intersect(set$cellIds, cells_id)
      data$custom[cells_id %in% filtered_cells] <- i
  }
```